local-tests/test_catch_up.py

```diff
@@ -0,0 +1,75 @@
+#!/bin/env python
+import os
+import sys
+from os.path import abspath, join
+from time import sleep
+
+from chainrunner import Chain, Seq, generate_keys
+
+# Path to working directory, where chainspec, logs and nodes' dbs are written:
+workdir = abspath(os.getenv('WORKDIR', '/tmp/workdir'))
+# Path to the aleph-node binary (important use short-session feature):
+binary = abspath(os.getenv('ALEPH_NODE_BINARY', join(workdir, 'aleph-node')))
+
+
+def check_finalized(nodes):
+    results = [node.highest_block() for node in nodes]
+    highest, finalized = zip(*results)
+    print('Blocks seen by nodes:')
+    print('  Highest:   ', *highest)
+    print('  Finalized: ', *finalized)
+
+    return finalized
+
+
+phrases = [f'//{i}' for i in range(6)]
+keys = generate_keys(binary, phrases)
+all_accounts = list(keys.values())
+chain = Chain(workdir)
+print('Bootstraping the chain with binary')
+chain.bootstrap(binary,
+                all_accounts[:4],
+                accounts=all_accounts[4:],
+                sudo_account_id=keys[phrases[0]],
+                chain_type='local')
+
+chain.set_flags(port=Seq(30334),
+                ws_port=Seq(9944),
+                rpc_port=Seq(9933),
+                unit_creation_delay=200,
+                execution='Native')
+
+chain.set_flags_validator('validator')
+
+print('Starting the chain')
+chain.start('aleph')
+
+print('Waiting 30s')
```

Comment:
Recently while running similar small tests locally I noticed that sometimes, at the very beginning of the chain life, the finalization is kinda lazy and lags at 0 until around block 100. Do you think that could be a problem here?

Reply:
I don't think the test will fail because of that but probably will not test fully what we want (catching up from some already finalized state)

Reply:
Then maybe some bigger delay (90s?) is worth considring here?

Reply:
set to 90s